### PR TITLE
8316389: Open source few AWT applet tests

### PR DESCRIPTION
**Pull request by azvegint**
Open sourcing few tests:

java/awt/Frame/FrameRepackTest.java
java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java
java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java
java/awt/Frame/WindowMoveTest.java

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316389](https://bugs.openjdk.org/browse/JDK-8316389): Open source few AWT applet tests (**Bug** - P4)

### Reviewers
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Committer) ⚠️ Review applies to [edf8f551](https://git.openjdk.org/jdk/pull/15787/files/edf8f5519cd5a693caeed48f18b20f01955b9903)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer) ⚠️ Review applies to [57cfd0bd](https://git.openjdk.org/jdk/pull/15787/files/57cfd0bd8468268357531a68b706f7c12cb2b2b2)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15787/head:pull/15787` \
`$ git checkout pull/15787`

Update a local copy of the PR: \
`$ git checkout pull/15787` \
`$ git pull https://git.openjdk.org/jdk.git pull/15787/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15787`

View PR using the GUI difftool: \
`$ git pr show -t 15787`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15787.diff">https://git.openjdk.org/jdk/pull/15787.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15787#issuecomment-1723333094)

### CODE REVIEW

File: test/jdk/java/awt/Frame/FrameRepackTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.SwingUtilities;
+import java.awt.BorderLayout;
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Frame/FrameRepackTest.java:25]: awt imports first here and others also.

File: test/jdk/java/awt/Frame/FrameRepackTest.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @key headful
+ * @summary Test dynamically changing frame component visibility and repacking
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual FrameRepackTest
+*/
```
Comment (DamonGuy) [inline test/jdk/java/awt/Frame/FrameRepackTest.java:44]:
```suggestion
 * @summary Test dynamically changing frame component visibility and repacking
 * @library /java/awt/regtesthelpers
 * @build PassFailJFrame
 * @run main/manual FrameRepackTest
 */
```

**Review by DamonGuy — APPROVED** [review]
Changes look good. One minor asterisk alignment found

**Review by kumarabhi006 — APPROVED** [review]

File: test/jdk/java/awt/Frame/FrameRepackTest.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+/*
+ * @test
+ * @key headful
+ * @summary Test dynamically changing frame component visibility and repacking
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual FrameRepackTest
+ */
+
+public class FrameRepackTest  {
```
Comment (turbanoff) [inline test/jdk/java/awt/Frame/FrameRepackTest.java:46]:
```suggestion
public class FrameRepackTest {
```

File: test/jdk/java/awt/Frame/FrameRepackTest.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    public FrameRepack() {
+        super("FrameRepack");
+
+        // create the menubar
+        MenuBar menubar = new MenuBar();
+        this.setMenuBar(menubar);
+        // create the options
+        Menu flip = new Menu("Flip");
+        MenuItem mi;
+        flip.addSeparator();
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameRepackTest.java:116]: The separator as the first item in a drop-down menu looks weird and confusing, I propose removing it.

File: test/jdk/java/awt/Frame/FrameRepackTest.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+        setLayout(new BorderLayout(2, 2));
+        setBackground(Color.green);
+
+        // north panel is always displayed
+        Panel north = new Panel();
+        north.setBackground(Color.red);
+        north.setLayout(new BorderLayout(2, 2));
+        north.add("North", new Label(" This panel is always displayed"));
+        north.add("Center", new Label(" it is a test "));
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameRepackTest.java:134]:
```suggestion
        north.add("North", new Label("This panel is always displayed"));
        north.add("Center", new Label("it is a test "));
```
Is the space used to offset the text from the edge?

File: test/jdk/java/awt/Frame/FrameRepackTest.java [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+        pack();
+    }
+
+    public void actionPerformed(ActionEvent evt) {
+        if (evt.getSource() instanceof MenuItem) {
+            if (evt.getActionCommand().equals("visible")) {
+                south.setVisible(true);
+                pack();
+                setVisible(true);
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameRepackTest.java:169]: Nothing should change if you remove `setVisible(true)` here, the frame is never hidden, is it?

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+ * @build PassFailJFrame
+ * @run main/manual FrameResizeTest_1
+ */
+
+public class FrameResizeTest_1 {
+
+    private static final String INSTRUCTIONS = """
+        To the right of this frame is an all-white 200x200 frame.
+
+        This is actually a white canvas component of the frame.
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java:45]:
```suggestion
        This is actually a white canvas component in the frame.
```

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+        The frame itself is red.
+        The red should never show.
+        In particular, after you resize the frame, you should see all white and no red.
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java:48]: If I resize the frame quickly, I see some areas in red, I guess it's expected…

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("FrameResizeTest_1 Instructions")
+                .instructions(INSTRUCTIONS)
+                .testTimeOut(5)
+                .rows(9)
+                .columns(45)
+                .build();
+
+        SwingUtilities.invokeAndWait(() -> {
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java:62]:
```suggestion
        EventQueue.invokeAndWait(() -> {
```
Using `EventQueue` is more appropriate for AWT components.

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+
+        setLayout( new GridBagLayout() );
+
+        GridBagConstraints c = new GridBagConstraints();
+        c.fill = GridBagConstraints.BOTH;
+        c.weightx = 1;
+        c.weighty = 1;
+
+        Container dumbc = new DumbC();
+        add( dumbc, c );
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java:95]:
```suggestion
        add(dumbc, c);
```

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+
+        Panel dump = new DumbPanel();
+        add( dump, c );
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java:98]:
```suggestion
        add(dump, c);
```

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+
+        setSize( new Dimension( 300, 300));
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java:100]:
```suggestion
        setSize(new Dimension(300, 300));
```
or directly `(300, 300)` without creating the `Dimension` object.

File: test/jdk/java/awt/Frame/WindowMoveTest.java [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+        }
+    }
+}
+
+class WindowMove extends Frame implements WindowListener {
+    static final Rectangle expectedBounds =
+            new Rectangle(100, 100, 300, 300);
+
+    static CountDownLatch latch = new CountDownLatch(1);
+    static volatile String failMessage = null;
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/WindowMoveTest.java:69]: It does not need to be `volatile`, using the latch establishes the _happens before_ relation between the EDT and the main thread.

File: test/jdk/java/awt/Frame/WindowMoveTest.java [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+
+        EventQueue.invokeAndWait(() -> frame = new WindowMove());
+
+        robot.waitForIdle();
+        robot.delay(1000);
+
+        EventQueue.invokeAndWait(() ->
+                frame.dispatchEvent(new WindowEvent(frame, WindowEvent.WINDOW_CLOSING)));
+
+        WindowMove.latch.await();
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/WindowMoveTest.java:56]: Add a timeout for `await` just in case? However, jtreg will also handle timeout.

**Review by aivanov-jdk — COMMENTED** [review]
Do we add `@Override` annotations?

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+        passFailJFrame.awaitAndCheck();
+    }
+}
+
+class FrameResize_2 extends Frame {
+
+    FrameResize_2() {
+        super("FrameResize_2");
+
+        setLayout( new GridBagLayout() );
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java:87]:
```suggestion
        setLayout(new GridBagLayout());
```

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    public void paint(Graphics g) {
+        Dimension d = getSize();
+        g.setColor(getBackground());
+        g.fillRect(0, 0, d.width, d.height);
+    }
+}
+
+class DumbC extends Container {
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java:118]:
```suggestion
class DumbContainer extends Container {
```
Similarly, to `DumbPanel`?

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+        super("FrameResize_2");
+
+        setLayout( new GridBagLayout() );
+
+        GridBagConstraints c = new GridBagConstraints();
+        c.fill = GridBagConstraints.BOTH;
+        c.weightx = 1;
+        c.weighty = 1;
+
+        Container dumbc = new DumbC();
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java:94]: Maybe change names of `dumbc` and `dump` to `container` and `panel`?

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java (hunk 8 above)
Comment (azvegint) [reply test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java:48]: Updated the instructions slightly to reflect this.

File: test/jdk/java/awt/Frame/FrameRepackTest.java (hunk 5 above)
Comment (aivanov-jdk) [reply test/jdk/java/awt/Frame/FrameRepackTest.java]: Yeah, it was it. Now the text is flushed to the left. Fine either way.

File: test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java (hunk 9 above)
Comment (aivanov-jdk) [reply test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_1.java:64]: Yeah, yet `JFrame` is hidden away inside `PassFailJFrame`.

The test itself uses AWT components, therefore it's reasonable not to use `SwingUtilities`. Functionally, there's no change, so I'm nitpicking.

File: test/jdk/java/awt/Frame/WindowMoveTest.java [hunk 18]
```diff
@@ -53,7 +54,9 @@ public static void main(String[] args) throws Exception {
         EventQueue.invokeAndWait(() ->
                 frame.dispatchEvent(new WindowEvent(frame, WindowEvent.WINDOW_CLOSING)));
 
-        WindowMove.latch.await();
+        if (!WindowMove.latch.await(10, TimeUnit.SECONDS)) {
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/Frame/WindowMoveTest.java:57]: I think 10 seconds is too long… We usually use 2 seconds for various delays.

**Review by aivanov-jdk — APPROVED** [review]

File: test/jdk/java/awt/Frame/WindowMoveTest.java (hunk 18 above)
Comment (prrace) [reply test/jdk/java/awt/Frame/WindowMoveTest.java:57]: I agree, If it doesn't happen in much less than 10 seconds probably it'll never happen.

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**